examples/traffic_histories_vehicle_replacement.py

```diff
@@ -61,27 +61,42 @@ def act(self, obs: Observation) -> Tuple[float, float]:
         if not exp:
             return (0.0, 0.0)
         cur_state = obs.ego_vehicle_state
+        exp_state = exp.ego_vehicle_state
 
         assert math.isclose(
-            cur_state.heading, exp["heading"], abs_tol=1e-2
-        ), f'vid={self._vehicle_id}: {cur_state.heading} != {exp["heading"]} @ {obs_time}'
+            cur_state.heading, exp_state.heading, abs_tol=1e-2
+        ), f"vid={self._vehicle_id}: {cur_state.heading} != {exp_state.heading} @ {obs_time}"
         # Note: the other checks can't be as tight b/c we lose some accuracy (due to angular acceleration)
         # by converting the acceleration vector to a scalar in the observation script,
         # which compounds over time throughout the simulation.
         assert math.isclose(
-            cur_state.speed, exp["speed"], abs_tol=0.2
-        ), f'vid={self._vehicle_id}: {cur_state.speed} != {exp["speed"]} @ {obs_time}'
+            cur_state.speed, exp_state.speed, abs_tol=0.2
+        ), f"vid={self._vehicle_id}: {cur_state.speed} != {exp_state.speed} @ {obs_time}"
         assert math.isclose(
-            cur_state.position[0], exp["ego_pos"][0], abs_tol=2
-        ), f'vid={self._vehicle_id}: {cur_state.position[0]} != {exp["ego_pos"][0]} @ {obs_time}'
+            cur_state.position[0], exp_state.position[0], abs_tol=2
+        ), f"vid={self._vehicle_id}: {cur_state.position[0]} != {exp_state.position[0]} @ {obs_time}"
         assert math.isclose(
-            cur_state.position[1], exp["ego_pos"][1], abs_tol=2
-        ), f'vid={self._vehicle_id}: {cur_state.position[1]} != {exp["ego_pos"][1]} @ {obs_time}'
+            cur_state.position[1], exp_state.position[1], abs_tol=2
+        ), f"vid={self._vehicle_id}: {cur_state.position[1]} != {exp_state.position[1]} @ {obs_time}"
 
         # Then get and return the next set of control inputs
         atime = self._rounder(obs_time + self._fixed_timestep_sec)
-        data = self._data.get(atime, {"acceleration": 0, "angular_velocity": 0})
-        return (data["acceleration"], data["angular_velocity"])
+        next_obs = self._data.get(atime)
+        if not next_obs:
+            return (0.0, 0.0)
```

Comment:
Should this ever happen?

Reply:
Yes I think so, perhaps on the last timestep? This guard was in the original script, so I'm assuming it was there for a reason.